src/lfs/server_zone.lua

```diff
@@ -0,0 +1,42 @@
+local module = ...
+
+local function process(request)
+  if request.method == "GET" then
+    if request.query then
+      request.query.zone = request.query.zone or "all"
+    end
+    local body = {}
+    if request.query.zone == "all" then
+      local sensors = require("sensors")
+      for i, sensor in pairs(sensors) do
+        table.insert(body, { zone = sensor.zone, state = sensor.state })
+      end
+    else
+      body = { {
+        zone = request.query.zone,
+        state = gpio.read(require("zone_to_pin")(request.query.zone))
+      } }
+    end
+    return sjson.encode(body)
+  end
+
+  if request.contentType == "application/json" then
+    if request.method == "PUT" then
+      if request.body[1] then
+        local ret = {}
+        for i in pairs(request.body) do
+          table.insert(ret, require("switch")(request.body[i]))
+        end
+        return sjson.encode(ret)
+      else
+        return sjson.encode(require("switch")(request.body))
+      end
+    end
+  end
+end
+
+return function(request)
+  package.loaded[module] = nil
+  module = nil
+  return process(request)
+end
```

Comment:
Ideally we would not need this copied file and could just enhance `server_device.lua` to do the job with either param key.

Reply:
Agreed, will remove. Was useful in testing mainly.

Reply:
done